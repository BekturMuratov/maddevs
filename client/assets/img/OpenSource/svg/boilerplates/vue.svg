<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>