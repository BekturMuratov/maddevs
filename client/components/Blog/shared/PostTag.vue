<template>
  <NuxtLink
    v-if="tag"
    :event="disabled ? '' : 'click'"
    :to="link"
    class="post-tag"
    :class="`post-tag--${theme}-theme`"
    data-testid="test-tag"
  >
    {{ tag }}
  </NuxtLink>
</template>

<script>
import linkResolver from '@/plugins/link-resolver.js'
import convertStringToSlug from '@/helpers/convertStringToSlug'

export default {
  name: 'PostTag',
  props: {
    tag: {
      type: String,
      default: '',
    },

    disabled: {
      type: Boolean,
      default: false,
    },

    theme: {
      type: String,
      default: 'light',
    },
  },

  computed: {
    link() {
      return linkResolver({ type: 'tag', uid: convertStringToSlug(this.tag) })
    },
  },
}
</script>

<style scoped lang="scss">
.post-tag {
  white-space: nowrap;
  font-size: 14px;
  line-height: 18px;
  letter-spacing: -0.4px;
  padding: 8px 16px;
  border-radius: 2px;
  transition: 0.2s;
  &--white-theme {
    background-color: $bgcolor--white-primary !important;
    color: $text-color--black !important;
  }
  &--light-theme {
    background-color: $bgcolor--silver !important;
    color: $text-color--black !important;
  }
  &--dark-theme {
    background-color: $text-color--quote-box !important;
    color: $text-color--white-transparent !important;
  }
}
</style>
