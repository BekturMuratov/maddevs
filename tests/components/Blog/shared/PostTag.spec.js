import { render } from '@testing-library/vue'
import PostTag from '@/components/Blog/shared/PostTag'

const TAG_NAME = 'post-tag'
const THEME_COLOR = 'very-dark'

const stubs = ['NuxtLink']

describe('post tag component', () => {
  it('is a Vue instance', () => {
    const { container, getByTestId } = render(PostTag, {
      stubs,
      props: {
        tag: TAG_NAME,
      },
    })

    expect(getByTestId('test-tag').innerHTML.trim()).toBe(TAG_NAME)
    expect(container).toMatchSnapshot()
  })

  it('to be render with dark theme', () => {
    const { html } = render(PostTag, {
      stubs,
      props: {
        tag: TAG_NAME,
        theme: THEME_COLOR,
      },
    })

    expect(html()).toContain(THEME_COLOR)
  })
})
