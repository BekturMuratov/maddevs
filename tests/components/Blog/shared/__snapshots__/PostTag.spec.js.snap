// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post tag component is a Vue instance 1`] = `
<div>
  <nuxtlink
    class="post-tag light"
    to="/blog/tag/posttag/"
  >
    <span
      class="tag"
      data-testid="test-tag"
    >
      
    post-tag
  
    </span>
  </nuxtlink>
</div>
`;
